forgerock-openbanking-uk-aspsp-rs/forgerock-openbanking-uk-aspsp-rs-gateway/forgerock-openbanking-uk-aspsp-rs-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/rs/api/payment/verifier/DetachedJwsVerifier.java

```diff
@@ -100,26 +103,30 @@ public void verifyDetachedJws(String detachedJws, OBVersion obVersion, HttpServl
         }
     }
 
-    private boolean isB64ClaimHeaderSetToFalse(String jwsDetachedSignature, String payload) throws OBErrorException, ParseException {
+
+    /**
+     * Returns true if the b64 header is true. This indicates that the base 64 encoded payload is included in the jws.
+     * @param jwsDetachedSignature
+     * @param payload
+     * @return
+     * @throws OBErrorException
+     * @throws ParseException
+     */
+    private boolean isBase64Encoded(String jwsDetachedSignature, String payload) throws OBErrorException,
+            ParseException {
         String jwsSerialized = rebuildJWS(jwsDetachedSignature, payload);
         log.debug("The reconstructed JWS from the detached signature: {}", jwsSerialized);
         SignedJWT jws = (SignedJWT) JWTParser.parse(jwsSerialized);
-        if (jws.getHeader().getCustomParam("b64") != null) {
-            log.debug("B64 claim header in the detached signature: {}", jws.getHeader().getCustomParam("b64"));
-            Object b64Param = jws.getHeader().getCustomParam("b64");
-            if (!(b64Param instanceof Boolean)) {
-                return false;
-            }
-            return ((Boolean)b64Param) == false;
-        }
-        return false;
+        boolean isBase64Encoded = jws.getHeader().isBase64URLEncodePayload();
+        return isBase64Encoded;
```

Comment:
You could get rid of `isBase64Encoded` and make it a one line return like you have in the method below :-)

Reply:
Have done that in latest refactoring